packages/remix-dev/config.ts

```diff
@@ -462,7 +479,22 @@ export async function resolveConfig(
   let pkgJson = await PackageJson.load(rootDirectory);
   let deps = pkgJson.content.dependencies ?? {};
 
-  if (userEntryServerFile) {
+  let isSpaMode = appConfig.ssr === false;
+  if (isSpaMode) {
+    // This is a super-simple default since we don't need streaming in SPA mode.
+    // We can include this in a remix-spa template, but right now `npx remix reveal`
+    // will still expose the streaming template since that command doesn't have
+    // access to the `ssr:false` flag in the vite config (the streaming template
+    // works just fine so maybe instea dof having this we _only have this version
+    // in the template...).  We let users manage an entry.server file in SPA mode
+    // so they can de ide if they want to hydrate the full document or just an
+    // embedded `<div id="app">` or whatever.
+    entryServerFile = "entry.server.spa.tsx";
+
+    // TODO: Should we check the rest of the server related configs and throw
+    // errors if present?  We need to be able to create and run a request handler
+    // to generate index.html, so we need _some_ aspect of "server" available.
```

Comment:
@pcattori @markdalgleish Curious to get your thoughts here - worth throwing build-time errors for a subset of configs that don't make sense in a SPA world?

Reply:
I think it makes sense to throw when resolving the plugin config.

Reply:
👍 I'll probably ping you or @pcattori to make sure I come up with the right list of configs to throw on

Reply:
It's only really `unstable_serverBundles` - added a warning in https://github.com/remix-run/remix/pull/8338/commits/b80ca86b5b70d1b9464d5f8233361bfec6e130c1.  Went with a warning since it feels weird to totally break a build if someone wants to just quickly try out SPA mode.  Instead we just ignore the `unstable_serverBundles` config